src/python/pants/goal/run_tracker.py

```diff
@@ -390,3 +395,29 @@ def shutdown_worker_pool(self):
     N.B. This exists only for internal use and to afford for fork()-safe operation in pantsd.
     """
     SubprocPool.shutdown(self._aborted)
+
+  def report_target_info(self, scope, target, key, val):
+    """
+    Add target information to run_info under target_data
+    :param string scope: The running scope
+    :param string target: The target that we want to store info for
+    :param list of keys key: The key for the info being stored
+    :param dict or string val: The value of the info being stored
+    """
+    def insert_key_value(keys, val_item, index):
```

Comment:
A comment explaining this function would be good...
```
# Recursively constructs a nested dict with keys made up of the components of the given key list.
```
...or maybe that explanation should go in the docstring for `report_target_info`?